consensus/ibft/src/test/java/tech/pegasys/pantheon/consensus/ibft/jsonrpc/methods/IbftGetSignerMetricsTest.java

```diff
@@ -0,0 +1,269 @@
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.SignerMetricResult;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class IbftGetSignerMetricsTest {
+
+  private final JsonRpcParameter jsonRpcParameter = new JsonRpcParameter();
+  private final String IBFT_METHOD = "ibft_getSignerMetrics";
+  private final String JSON_RPC_VERSION = "2.0";
+  private IbftGetSignerMetrics method;
+
+  private BlockchainQueries blockchainQueries;
+  private IbftBlockInterface ibftBlockInterface;
+
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setup() {
+    blockchainQueries = mock(BlockchainQueries.class);
+    ibftBlockInterface = mock(IbftBlockInterface.class);
+    method = new IbftGetSignerMetrics(ibftBlockInterface, blockchainQueries, jsonRpcParameter);
+  }
+
+  @Test
+  public void returnsCorrectMethodName() {
+    assertThat(method.getName()).isEqualTo(IBFT_METHOD);
+  }
+
+  @Test
+  public void exceptionWhenInvalidStartBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 0");
+
+    method.response(request);
+  }
+
+  @Test
+  public void exceptionWhenInvalidEndBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("1", "INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 1");
+
+    method.response(request);
+  }
+
+  @Test
+  public void getSignerMetricsWhenNoParams() {
+
+    Long startBlock = 1L;
+    Long endBlock = 3L;
+
+    when(blockchainQueries.headBlockNumber()).thenReturn(endBlock);
+
+    List<SignerMetricResult> signerMetricResultList = new java.util.ArrayList<>();
+
+    LongStream.range(startBlock, endBlock)
+        .forEach(value -> signerMetricResultList.add(generateBlock(value)));
+
+    final JsonRpcRequest request = requestWithParams();
+
+    final JsonRpcSuccessResponse response = (JsonRpcSuccessResponse) method.response(request);
+
+    LongStream.range(startBlock, endBlock)
+        .forEach(value -> verify(blockchainQueries).getBlockHeaderByNumber(value));
+
+    verify(blockchainQueries, atLeast(2)).headBlockNumber();
```

Comment:
I don't think there's any reason to assert these class to `blockchainQueries` - they're just implementation details that it needs to get the information.  We just check it comes up with the right answer.

Reply:
is removed